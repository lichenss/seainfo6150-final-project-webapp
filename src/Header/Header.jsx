import React from 'react';
import { useHistory } from 'react-router-dom'
import { Link } from 'react-router-dom';
import styles from './Header.module.css';

const Header = () => {
    const history = useHistory()

    return (
        <div>
            <header className={styles.header}>
                <div className={styles.navbar}>
                <button className={styles.logo} onClick={() => {history.push('/')}}></button>
                <div>
                <Link className={styles.catLink} to='/category/computer-science'>Computer Science</Link>
                <Link className={styles.catLink} to='/category/business'>Business</Link>
                <Link className={styles.catLink} to='/category/data-science'>Data Science</Link>
                <Link className={styles.catLink} to='/category/arts-and-humanities'>Arts and Humanities</Link>
                <Link className={styles.catLink} to='/category/physical-science-and-engineering'>Physical Science and Engineering</Link>
                <Link className={styles.catLink} to='/category/math-and-logic'>Math and Logic</Link>
                </div>
                    <div className={styles.links}>
                            <Link className={styles.barLink} to="/log-in">
                                Log In
                            </Link>                      
                            <Link className={styles.barLink} to="/sign-up">
                                Sign Up
                            </Link>
                        
                    </div>
                </div>
            </header>
        </div>
    );
};

export default Header;