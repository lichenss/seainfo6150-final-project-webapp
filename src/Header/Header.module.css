.header {
    background-color: #f7f3f3;
    top: 0;
    left: auto;
    right: 0;
    position: fixed;
    width: 100%;
    z-index: 1;
    font-family: 'Source Sans Pro', serif;
}
.logo {
    background-color: transparent;
    background: url("../images/logo.png");
    border:none;
    background-repeat: no-repeat;
    background-size: contain;
    width: 3rem;
    height: 3rem;
    padding:0;
    cursor: pointer;
    outline:none;
}
.navbar {
    display: flex;
    position: relative;
    align-items: center;
    justify-content: space-evenly;
    padding-left: 24px;
    padding-right: 24px;
    height: 70px;
}

.backContainer {
    flex:1;
}

.barLink{
    color:#17488D;
    font-weight: 600;
    line-height: 1.6;
    font-size: 1.5rem;
    margin-left: 24px;
    text-decoration: none;
}

.links {
    font-family:'Roboto', sans-serif;
    flex:1;
    display: flex;
    justify-content: flex-end;
}

.categorylinks {
    display: flex;
}
.catLink{
    color:#17488D;
    line-height: 1.6;
    font-size: 1rem;
    margin: 10px;
    text-decoration: none;
}

@media screen and (min-width:600px){
    .barLink{
        font-size: 1.0rem;
    }
}
@media screen and (min-width:900px){
    .barLink{
        font-size: 1.3rem;
    }
}
@media screen and (min-width:1200px){
    .barLink{
        font-size: 1.5rem;
    }
}