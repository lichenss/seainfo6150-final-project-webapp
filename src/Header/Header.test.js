import React from "react";
import Header from "./Header";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
describe("Header tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <Header />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });