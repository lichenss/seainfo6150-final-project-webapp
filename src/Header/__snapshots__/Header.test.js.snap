// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders correctly 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="navbar"
      >
        <button
          class="logo"
        />
        <div>
          <a
            class="catLink"
            href="/category/computer-science"
          >
            Computer Science
          </a>
          <a
            class="catLink"
            href="/category/business"
          >
            Business
          </a>
          <a
            class="catLink"
            href="/category/data-science"
          >
            Data Science
          </a>
          <a
            class="catLink"
            href="/category/arts-and-humanities"
          >
            Arts and Humanities
          </a>
          <a
            class="catLink"
            href="/category/physical-science-and-engineering"
          >
            Physical Science and Engineering
          </a>
          <a
            class="catLink"
            href="/category/math-and-logic"
          >
            Math and Logic
          </a>
        </div>
        <div
          class="links"
        >
          <a
            class="barLink"
            href="/log-in"
          >
            Log In
          </a>
          <a
            class="barLink"
            href="/sign-up"
          >
            Sign Up
          </a>
        </div>
      </div>
    </header>
  </div>
</div>
`;
